tests/libgit/test_libgit_spec.lua

```diff
@@ -1,190 +1,180 @@
-
-  end)
-
-  it("Tests setting SSL_Cipher and User_Agent ", function()
-    -- We simply call the setting functions.
-    -- If there will be any error, The program
-    -- will shutdown itself
-
-    lib.luagit2_SET_USER_AGENT(User_Agent)
-    lib.luagit2_SET_SSL_CIPHERS(SSL_Cipher)
-  end)
-
-  it("Tests Setting Object's Cache Limit ", function()
-    -- We call the functions with possible
-    -- git object types. If there will be any
-    -- error the program will shut down itself.
-    local type_commit_obj = lib.luagit2_object_string2type("commit")
-    local type_tree_obj = lib.luagit2_object_string2type("tree")
-    local type_blob_obj = lib.luagit2_object_string2type("blob")
-    local type_tag_obj = lib.luagit2_object_string2type("tag")
-    local type_OFS_DELTA = lib.luagit2_object_string2type("OFS_DELTA")
-    local type_REF_DELTA = lib.luagit2_object_string2type("REF_DELTA")
-
-    lib.luagit2_SET_CACHE_OBJECT_LIMIT(type_commit_obj,Object_Cache_Limit)
-    lib.luagit2_SET_CACHE_OBJECT_LIMIT(type_tree_obj,Object_Cache_Limit)
-    lib.luagit2_SET_CACHE_OBJECT_LIMIT(type_blob_obj,Object_Cache_Limit)
-    lib.luagit2_SET_CACHE_OBJECT_LIMIT(type_tag_obj,Object_Cache_Limit)
-    lib.luagit2_SET_CACHE_OBJECT_LIMIT(type_OFS_DELTA,Object_Cache_Limit)
-    lib.luagit2_SET_CACHE_OBJECT_LIMIT(type_REF_DELTA,Object_Cache_Limit)
-  end)
-
+    
+    local luagit2 = require("luagit2")
+    local mwindow_size = 1234
+    local mwindow_mapped_limit = 1234
+    local cache_max_size = 2048
+    local template_path = "./Example/Template/path/"
+    local windows_share_mode = 99999
+    local enable = 1
+    local disable = 0
+    local search_paths = {
+        system_lvl = "./System_level/Search_Path/",
+        global_lvl = "./Global_level/Search_Path/",
+        xdg_lvl = "./XDG_level/Search_Path/",
+        program_data_lvl = "./Program_data_level/Search_Path/",
+    }
+    local alternate_path = "./Some/Different/Path/"
+    
+    setup(function()
+        -- initialize libgit2's global state and threading
+        luagit2.init()
+        
+        -- setup different variables
+        luagit2.set_mwindow_size(mwindow_size)
+        luagit2.set_mwindow_mapped_limit(mwindow_mapped_limit)
+        luagit2.set_search_path("system", search_paths.system_lvl)
+        luagit2.set_search_path("global", search_paths.global_lvl)
+        luagit2.set_search_path("xdg", search_paths.xdg_lvl)
+        luagit2.set_search_path("programdata", search_paths.program_data_lvl)
+        luagit2.set_cache_max_size(cache_max_size)
+        luagit2.set_template_path(template_path)
+        luagit2.set_windows_sharemode(windows_share_mode)
+    end)
+    
+    teardown(function()
+        -- close threading and global state.
+        luagit2.shutdown()
+    end)
+    
+    it("Prints used libgit2 version", function ()
+        local major, minor, rev = luagit2.libgit2_version()
+        print("Used libgit2 version is : " .. major .. "." .. minor .. "." .. rev)
```

Comment:
awesome! this is much better 🎉

Reply:
thankyou :smile: I looks very good and feels very nice using in lua side.